hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoUpdateHandler.java

```diff
@@ -60,7 +60,7 @@ public void preHandle(@NonNull final PreHandleContext context, @NonNull final Cr
         if (targetAccountKeyMustSign) {
             context.addNonPayerKey(updateAccountId);
         }
-        if (newAccountKeyMustSign && op.key() != null) {
+        if (newAccountKeyMustSign && op.hasKey()) {
```

Comment:
You'll have to use `hasKeyOrThrow()` to get rid of the warning here.

Reply:
hasKey shouldn't be able to throw or return null...

Reply:
yes Agreed. Fixed . Thanks Richard